community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -54,20 +53,101 @@
 TOT_REQ_CNT = 1000
 _MAINTENANCE_SBATCH_SCRIPT_PATH = dirs.custom_scripts / "perform_maintenance.sh"
 
-NodeStatus = Enum(
-    "NodeStatus",
-    (
-        "orphan",
-        "power_down",
-        "preempted",
-        "restore",
-        "resume",
-        "terminated",
-        "unbacked",
-        "unchanged",
-        "unknown",
-    ),
-)
+class NodeAction(Protocol):
+    def apply(self, nodes:List[str]) -> None:
+        ...
+
+    def __hash__(self):
```

Comment:
I don't see a good reason to provide an implementation of `__hash__` and `__eq__`,
as well as using `NodeAction` as an "abstract" class.

Reply:
What would be the remaining utility of having `NodeAction` at all if the inheritance is removed?

Reply:
The utility is in interface (type checking)